packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart

```diff
@@ -155,8 +160,13 @@ class MethodChannelShare extends SharePlatform {
   }
 
   /// A wrapper of [MethodChannelShare._getFile] for multiple files.
-  Future<List<XFile>> _getFiles(List<XFile> files) async =>
-      await Future.wait(files.map((entry) => _getFile(entry)));
+  Future<List<XFile>> _getFiles(
+          List<XFile> files, List<String>? fileNameOverrides) async =>
+      (fileNameOverrides == null)
+          ? await Future.wait(files.map((entry) => _getFile(entry)))
+          : await Future.wait(files.asMap().entries.map((entry) => _getFile(
+              entry.value,
+              nameOverride: fileNameOverrides[entry.key])));
```

Comment:
Need to look if this can be simplified as well.

Reply:
We could convert the function from an expression to a block body and then use a for loop, like so
```dart
  /// A wrapper of [MethodChannelShare._getFile] for multiple files.
  Future<List<XFile>> _getFiles(
          List<XFile> files, List<String>? fileNameOverrides) async {
    final List<XFile> retrievedFiles = [];
    for (var index = 0; index < files.length; index++) {
      final String? fileName = fileNameOverrides?.elementAt(index);
      retrievedFiles.add(await _getFile(files[index], nameOverride:  fileName));
    }
    return retrievedFiles;
  }
```
I'm not completely sure if this would be simpler though.

Reply:
Yes, I think so, I much prefer to use indexes as well.

Reply:
I pushed a refactor to your fork, I hope you don't mind!